xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -1639,7 +1639,12 @@ const CFileItemList& CGUIMediaWindow::CurrentDirectory() const
 
 bool CGUIMediaWindow::WaitForNetwork() const
 {
-  if (g_application.getNetwork().IsAvailable())
+  // temporary hack to revert behaviour prior
+  // to IsAvailable() remove to proceed with testing
+  // (with no network configured)
+  return true;
```

Comment:
@MartijnKaijser 

I was deliberately keeping some commits to squash or this one "open" - to be reminded that there are some points to discuss.

This one is about this:
- there was/is IsAvailable() function - which should have provided either status of available network  (up/down) or via bool parameter a blocking call (a loop with timeout) until network is up
- historically it wasn't used that (latter) way and with some other commit the wait parameter was removed and function cramped to simple yes/no return value
- even though the POSIX platforms (implementation in LinuxNetwork) now have event triggering network watcher - so IsAvailable with wait = yes - could be reverted and finally work as intended originally, there is currently no user of that (either Kodi's code or distributions using Kodi)

- also with removing of 'wait' parameter and necessary body update the function was simply returning true if any IF was red from system (regardless if it's state)

that means that going up in the chain its caller CGUIMediaWindow::WaitForNetwork() is since that particular commit not serving it's purpose and either will be a 'buggy' blocker to almost any user action invoking media content, or a dummy useless call up the stack returning true

so two options:
1. the said commit will be reverted and we make IsAvailable() to be blocking (with wait=yes) and sleeping waiting for event from network stack (that system got connected IF)
or
2. CGUIMediaWindow::WaitForNetwork() will be inlined with return true;

(case 1 would need to implement ~5 lines into w32 platform code - to wait & provide the tick on network reconfiguration / status change)